streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -224,6 +224,9 @@ void initializeStoreOffsetsFromCheckpoint(final boolean storeDirIsEmpty) {
             for (final StateStoreMetadata store : stores.values()) {
                 if (store.changelogPartition == null) {
                     log.info("State store {} is not logged and hence would not be restored", store.stateStore.name());
+                } else if (!store.stateStore.persistent()) {
+                    log.info("Initializing to the starting offset for changelog {} of in-memory state store {}",
+                             store.changelogPartition, store.stateStore.name());
```

Comment:
Don't we need to call `store.setOffset(null)` or this case?

Reply:
Just wondering about https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java#L769-L802 (can't comment below).

For this case, if should hold that `store.offset() == changelogOffsetFromCheckpointedOffset(loadedCheckpoints.remove(store.changelogPartition))` ?

Or maybe `>=`?

Should we add a sanity check? (Not related to this PR itself actually. -- Just wondering.)

Reply:
> Don't we need to call store.setOffset(null) or this case?

Well, either it's a recycled task in which case no, we don't want to wipe out the existing offset, or it's a new task in which case it's initialized to `null` anyway.

I'm also not sure what you mean in the second comment. Did you maybe paste the link to the wrong code? (just guessing since you linked to that same code earlier in John's PR)

Reply:
Ups. Wrong link. This one: https://github.com/apache/kafka/pull/8996/files#diff-cc98a6c20f2a8483e1849aea6921c34dL251-L255

Reply:
> Well, either it's a recycled task in which case no, 

So a active->standby or standby->active conversion. I was wondering about corrupted tasks? So we don't call `initializeStoreOffsetsFromCheckpoint` when reviving a corrupted task?

Reply:
No, because we close the state manager completely and clear all this data before reviving a corrupted task.

I've had to re-convince myself of this several times already. Maybe we can also add a check that none of them are marked corrupted while initializing offsets